partiql-value/src/datum.rs

```diff
@@ -310,6 +353,24 @@ impl Iterator for DatumSeqOwnedIterator {
         match self {
             DatumSeqOwnedIterator::List(l) => l.next(),
             DatumSeqOwnedIterator::Bag(b) => b.next(),
+            DatumSeqOwnedIterator::Dynamic(d) => d.next(),
         }
     }
 }
+
+// TODO remove
+pub struct DynIntoIterator(vec::IntoIter<Value>);
+
+impl Iterator for DynIntoIterator {
+    type Item = Value;
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        self.0.next()
+    }
+
+    #[inline]
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.0.size_hint()
+    }
+}
```

Comment:
(feel free to mark as resolved if this'll be removed in a future PR)
A leftover TODO?

Reply:
Removed in #544 : https://github.com/partiql/partiql-lang-rust/pull/544/files#diff-bb8927121fdc73652316f60348ac039745ead27e428cd90574e410a50eb01ffbL360